actors/user/src/main/java/org/sunbird/user/actors/CertificateActor.java

```diff
@@ -1,83 +1,141 @@
+        }
     }
 
-  }
+    private Map<String, Object> getRequiredRequest(Map<String, Object> certAddReqMap) {
+        Certificate certificate = objectMapper.convertValue(certAddReqMap, Certificate.class);
+        return objectMapper.convertValue(certificate, Map.class);
+    }
 
-  /** This method validates the access-code of the certificate and retrieve certificate details.
-   * @param userRequest
-   */
-  private void getCertificate(Request userRequest) {
-    Map request = userRequest.getRequest();
-    String certificatedId = (String) request.get(JsonKey.CERT_ID);
-    String accessCode = (String) request.get(JsonKey.ACCESS_CODE);
-    Map<String, Object> responseDetails = getCertificateDetails(certificatedId);
-    if(responseDetails.get(JsonKey.ACCESS_CODE.toLowerCase()).equals(accessCode)) {
-      Map userResponse = new HashMap<String, Object>();
-      Response userResult = new Response();
-      Map recordStore = (Map<String, Object>) responseDetails.get(JsonKey.STORE);
-      userResponse.put(JsonKey.JSON,recordStore.get(JsonKey.JSON));
-      userResponse.put(JsonKey.PDF,recordStore.get(JsonKey.PDF));
-      userResult.put(JsonKey.RESPONSE,userResponse);
-      sender().tell(userResult, self());
-    } else {
-      ProjectLogger.log(
-              "CertificateActor:getCertificate: access code is incorrect : "
-                      + accessCode,
-              LoggerEnum.ERROR.name());
-      throw new ProjectCommonException(
-              ResponseCode.invalidParameter.getErrorCode(),
-              ProjectUtil.formatMessage(ResponseCode.invalidParameter.getErrorMessage(),JsonKey.ACCESS_CODE),
-              ResponseCode.CLIENT_ERROR.getResponseCode());
+    private static Timestamp getTimeStamp() {
+        return new Timestamp(Calendar.getInstance().getTime().getTime());
     }
-  }
 
-  private Map getCertificateDetails(String certificatedId) {
-    Map<String, Object> responseDetails = null;
-    Response response = cassandraOperation.getRecordById(JsonKey.SUNBIRD,JsonKey.USER_CERT,certificatedId);
-    Map<String, Object> record = response.getResult();
-    if(null != record && null != record.get(JsonKey.RESPONSE)) {
-      List responseList = (List) record.get(JsonKey.RESPONSE);
-      if(!responseList.isEmpty()) {
-        responseDetails = (Map<String, Object>) responseList.get(0);
-      } else {
-        ProjectLogger.log(
-                "CertificateActor:getCertificate: cert id is incorrect : "
-                        + certificatedId,
-                LoggerEnum.ERROR.name());
-        throw new ProjectCommonException(
-                ResponseCode.invalidParameter.getErrorCode(),
-                ProjectUtil.formatMessage(ResponseCode.invalidParameter.getErrorMessage(),JsonKey.CERT_ID),
-                ResponseCode.CLIENT_ERROR.getResponseCode());
-      }
+    private void assureUniqueCertId(String certificatedId) {
```

Comment:
We can make this method as reusable which will accept keyspace, table and id , based on that it will return boolean flag.
isIdentityPresent can be method name.